daemons/ipa-kdb/ipa_kdb_principals.c

```diff
@@ -2779,10 +2856,37 @@ static krb5_error_code ipadb_modify_principal(krb5_context kcontext,
     return kerr;
 }
 
+static krb5_error_code
+remove_virtual_str_attrs(krb5_context kcontext, krb5_db_entry *entry)
+{
+    char *str_attr_val;
+    krb5_error_code kerr;
+
+    kerr = krb5_dbe_get_string(kcontext, entry,
+                               OPT_PAC_TKT_CHKSUM_STR_ATTR_NAME,
+                               &str_attr_val);
+    if (kerr)
+        return kerr;
+
+    if (str_attr_val)
+        kerr = krb5_dbe_set_string(kcontext, entry,
+                                   OPT_PAC_TKT_CHKSUM_STR_ATTR_NAME,
+                                   NULL);
+
+    krb5_dbe_free_string(kcontext, str_attr_val);
+    return kerr;
+}
+
 krb5_error_code ipadb_put_principal(krb5_context kcontext,
                                     krb5_db_entry *entry,
                                     char **db_args)
 {
+    krb5_error_code kerr;
+
+    kerr = remove_virtual_str_attrs(kcontext, entry);
```

Comment:
My initial plan was to strip the `optional_pac_tkt_chksum` string attribute as part of `ipadb_get_ldap_mod_extra_data()`, but given the fact that in the krb5 API a string attribute can only be removed from an entry and not from the TL data attribute itself, it would have resulted in the initial entry to be modified too.

Since modifying the entry cannot be avoided, I thought it would be better to do it right at the beginning of the modification operation.